crates/e2e/src/node_proc.rs

```diff
@@ -161,11 +161,20 @@ where
                 e
             )
         })?;
-
-        // Wait for RPC port to be logged (it's logged to stderr):
         let stderr = proc.stderr.take().unwrap();
-        let port = find_substrate_port_from_output(stderr);
-        let url = format!("ws://127.0.0.1:{port}");
+        // Wait for RPC port to be logged (it's logged to stderr):
+        let port_str: String = match option_env!("WS_PORT") {
```

Comment:
Is it the case that the node that is spawned here does not print out the port to `stderr`, or are we just attaching to a different already running process (chopsticks?)

Reply:
Ok so you have clarified that chopsticks is an already running process.

In that case, I don't think we should be modifying `node_proc` in this way. It is intended to spin up a node from scratch and then return a client which can communicate with the node. So in this case it is spinning up a node for the test which is never used (because requests are sent to the chopsticks node via the overriding of the port).

What we could do instead is allow the user to specify the full url of a running process in the config. This would override the default behaviour of spawning a node for each test. e.g.

`#[ink_e2e::test(node_urk = "ws://127.0.0.1:8000")]`

Then in this case it wouldn't spawn the node_proc https://github.com/paritytech/ink/blob/6fa4343e5df8fd025179bceeec527ef8874ad78f/crates/e2e/macro/src/codegen.rs#L116

Reply:
Alternatively can use an environment variable, as per https://github.com/paritytech/ink/issues/1650.

However it may be good to have it in the attribute config, because this changes how e2e tests are written: usually they are expecting clean state on a chain and they don't affect each other. With a long running node the state is shared between tests so is not deterministic.